### ISSUE TITLE
golang

### ISSUE THREAD

**Issue by mikesmitty** [iuscommunity/wishlist#89]
### What package do you want?

go 1.7
### Why?

go 1.7 introduces the contexts library as well as speed improvements (10-35%) and reduced binary size.
### OS versions

I would like this package for both RHEL 6 and 7.
### Other

I'm already packaging the latest versions of go for RHEL/CentOS 6 and 7 and I'd be happy to offer my srpms or spec/patch files if you would like to include go in your repo.

**Comment by b-harper** [comment]
Hey @mikesmitty,

Thanks for taking the time to submit this request.  EPEL6 contains golang 1.5.1 and EL7 contains 1.6.3.   The goals of IUS is to provide newer versions of software found in EL.  EPEL is the appropriate repository for software not found in EL. We have made a few exceptions and packaged newer version of software found in EPEL.  This is only after the EPEL maintainers have been contacted and updating the package does not align with [EPEL guidelines][0], like [ABI changes or the config files need manual adjustments][1].

For EL6, I would recommend contacting the EPEL maintainers.

For EL7, there is a request to [rebase golang to 1.7][2].  Recently, Red Hat has been more open to bump versions than in the past, so I am not sure if they will proceed with that request.  I would want hold off on this request until we get some indication if/when Red Hat will update golang.  


[0]: https://fedoraproject.org/wiki/EPEL/GuidelinesAndPolicies
[1]: https://fedoraproject.org/wiki/EPEL/GuidelinesAndPolicies#A_major_version_update
[2]: https://bugzilla.redhat.com/show_bug.cgi?id=1365649

**Comment by carlwgeorge** [comment]
I just saw on the [epel-devel mailing list](https://lists.fedoraproject.org/archives/list/epel-devel@lists.fedoraproject.org/thread/KYZ3WI3LQLHZMRGDT5RJ7OR5V2L7X4NH/) that golang has a [pending update to golang-1.7.1-1.el6 in EPEL6](https://bodhi.fedoraproject.org/updates/FEDORA-EPEL-2016-1117d3c7f6).  Once that is available in their testing repository, I recommend trying it out and providing karma.

**Comment by carlwgeorge** [comment]
The [golang 1.7 update for EPEL6](https://bodhi.fedoraproject.org/updates/FEDORA-EPEL-2016-1117d3c7f6) has been pushed to stable.

**Comment by carlwgeorge** [comment]
I just noticed the [EL7 bug](https://bugzilla.redhat.com/show_bug.cgi?id=1365649) was changed the "Fixed In Version" field to `golang-1.7.4-3.el7` and has an `ON_QA` status.

**Comment by carlwgeorge** [comment]
The pending RHEL 7.4 release has golang 1.8.
```
# repoquery --enablerepo rhel-7-server-optional-beta-rpms --info golang

Name        : golang
Version     : 1.8
Release     : 4.el7
Architecture: x86_64
Size        : 12024752
Packager    : Red Hat, Inc. <http://bugzilla.redhat.com/bugzilla>
Group       : Unspecified
URL         : http://golang.org/
Repository  : rhel-7-server-optional-beta-rpms
Summary     : The Go Programming Language
Source      : golang-1.8-4.el7.src.rpm
Description :
The Go Programming Language.
```
Since this will be resolved soon by Red Hat we won't be adding it to IUS.
